interface/super/edit_globals.php

```diff
@@ -7,175 +7,74 @@
  * @author    Rod Roark <rod@sunsetsystems.com>
  * @author    Brady Miller <brady.g.miller@gmail.com>
  * @author    Ranganath Pathak <pathak@scrs1.org>
+ * @author    Robert Down <robertdown@live.com>
  * @copyright Copyright (c) 2010 Rod Roark <rod@sunsetsystems.com>
  * @copyright Copyright (c) 2016-2019 Brady Miller <brady.g.miller@gmail.com>
  * @copyright Copyright (c) 2019 Ranganath Pathak <pathak@scrs1.org>
+ * @copyright Copyright (c) 2019 Robert Down <robertdown@live.com>
  * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
  */
 
 
-require_once("../globals.php");
-require_once("../../custom/code_types.inc.php");
-require_once("$srcdir/acl.inc");
-require_once("$srcdir/globals.inc.php");
-require_once("$srcdir/user.inc");
-require_once(dirname(__FILE__)."/../../myportal/soap_service/portal_connectivity.php");
+require_once "../globals.php";
+require_once "../../custom/code_types.inc.php";
+require_once "$srcdir/acl.inc";
+require_once "$srcdir/globals.inc.php";
+require_once "$srcdir/user.inc";
+require_once "../../myportal/soap_service/portal_connectivity.php";
 
 use OpenEMR\Common\Crypto\CryptoGen;
 use OpenEMR\Common\Csrf\CsrfUtils;
-use OpenEMR\Common\Logging\EventAuditLogger;
-use OpenEMR\Core\Header;
-use OpenEMR\OeUI\OemrUI;
 use Ramsey\Uuid\Uuid;
+use OpenEMR\Admin\Service\SettingsService;
+use Symfony\Component\HttpFoundation\Request;
+
+/** @var Request */
+$request = Request::createFromGlobals();
+
+// Grab Twig environment and add the admin templates dir with "admin" namespace
+$twig = $GLOBALS['twig'];
+$twig->getLoader()->addPath("../../src/Admin/templates/globals", "admin");
+
+global $GLOBALS_METADATA;
+global $USER_SPECIFIC_TABS;
+global $USER_SPECIFIC_GLOBALS;
 
 // Set up crypto object
 $cryptoGen = new CryptoGen();
 
-$userMode = (array_key_exists('mode', $_GET) && $_GET['mode'] == 'user');
+$settingsService = new SettingsService($GLOBALS_METADATA, $USER_SPECIFIC_TABS, $USER_SPECIFIC_GLOBALS, $cryptoGen);
+
+/** @var bool $isUser */
+$isUser = ($request->query->get("user") === "user") ? true : false;
+$isUser = (array_key_exists('mode', $_GET) && $_GET['mode'] == 'user');
```

Comment:
won't the second $isUser overwrite the first $isUser?

Reply:
Yep, still some cleanup to do :)